apps/web/src/pages/EmployeeProfile/components/CareerDevelopmentSection/CareerDevelopmentSection.tsx

```diff
@@ -0,0 +1,503 @@
+        execCoachingStatus: execCoachingStatusToData(execCoachingStatus),
+        execCoachingInterest: execCoachingInterest ?? [],
+      },
+    })
+      .then((result) => {
+        if (result.data?.updateEmployeeProfile) {
+          toast.success(
+            intl.formatMessage({
+              defaultMessage:
+                "Career development preferences updated successfully!",
+              id: "ECAM6A",
+              description:
+                "Message displayed when a user successfully updates employee profile information",
+            }),
+          );
+          setIsEditing(false);
+          methods.reset(
+            {
+              organizationTypeInterest,
+              moveInterest,
+              mentorshipStatus,
+              mentorshipInterest,
+              execInterest,
+              execCoachingStatus,
+              execCoachingInterest,
+            },
+            { keepDirty: true },
+          );
+        } else {
+          handleError();
+        }
+      })
+      .catch(handleError);
+  };
+
+  const subtitle = intl.formatMessage({
+    defaultMessage:
+      "This section of your profile allows you to express interest in a variety of specific options related to recruitment, mentorship, and promotional opportunities.",
+    id: "Uj7TE2",
+    description:
+      "Describes the career development preferences of employee profile",
+  });
+
+  const organizationTypeInterestOptions: CheckboxOption[] = unpackMaybes(
+    careerDevelopmentOptions?.organizationTypeInterest,
+  ).map(({ value }) => ({
+    value,
+    label: intl.formatMessage(getOrganizationTypeInterest(value)),
+  }));
+
+  const moveInterestOptions: CheckboxOption[] = unpackMaybes(
+    careerDevelopmentOptions?.moveInterest,
+  ).map(({ value }) => ({
+    value,
+    label: intl.formatMessage(getMoveInterest(value)),
+  }));
+
+  const mentorshipStatusOptions: Radio[] = [
+    { value: MentorshipStatus.NOT_PARTICIPATING },
+    ...unpackMaybes(careerDevelopmentOptions?.mentorship),
+    { value: MentorshipStatus.MENTEE_AND_MENTOR },
+  ].map(({ value }) => ({
+    value,
+    label: intl.formatMessage(getMentorshipStatus(value)),
+  }));
+
+  const mentorshipInterestOptions: Radio[] = unpackMaybes(
+    careerDevelopmentOptions?.mentorship,
+  ).map(({ value }) => ({
+    value,
+    label: intl.formatMessage(getMentorshipInterest(value)),
+  }));
+
+  const execCoachingStatusOptions: Radio[] = [
+    { value: ExecCoachingStatus.NOT_PARTICIPATING },
+    ...unpackMaybes(careerDevelopmentOptions?.execCoaching),
+    { value: ExecCoachingStatus.LEARNING_AND_COACHING },
+  ].map(({ value }) => ({
+    value,
+    label: intl.formatMessage(getExecCoachingStatus(value)),
+  }));
+
+  const execCoachingInterestOptions: Radio[] = unpackMaybes(
+    careerDevelopmentOptions?.execCoaching,
+  ).map(({ value }) => ({
+    value,
+    label: intl.formatMessage(getExecCoachingInterest(value)),
+  }));
```

Comment:
Seems like we have a lot of loops here for only 3-4 items. Would it be better to just write these arrays directly in the prop?

Reply:
Moved in [6f560cb](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12680/commits/6f560cb4f502bfa0da754e1146a8c2fdb03d2855)